odoo/addons/base/tests/test_expression.py

```diff
@@ -1097,6 +1295,87 @@ def test_proper_combine_unit_leaves(self):
         expr = expression.OR([[], normal])
         self.assertEqual(expr, true)
 
+    def test_combine_simple_conditions(self):
+        # test that boolean leaves are properly handled in specific cases
+        false = Domain.FALSE
+        true = Domain.TRUE
+        normal = Domain('foo', '=', 'bar')
+        # OR and AND with empty list should return their zero value
+        self.assertEqual(Domain.OR(), false)
+        self.assertEqual(Domain.AND(), true)
+        # OR with single FALSE_LEAF
+        expr = Domain.OR(false)
+        self.assertEqual(expr, false)
```

Comment:
From a readability standpoint, all those useless assignments of `expr` suck.  I would prefer one-liners in this case, since they would look more tabular.
```suggestion
        self.assertEqual(Domain.OR(false), false)
```